plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -61,31 +81,48 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
 	if len(a.GetNamespace()) == 0 || a.GetKind().GroupKind() == api.Kind("Namespace") {
 		// if a namespace is deleted, we want to prevent all further creates into it
 		// while it is undergoing termination.  to reduce incidences where the cache
-		// is slow to update, we forcefully remove the namespace from our local cache.
-		// this will cause a live lookup of the namespace to get its latest state even
-		// before the watch notification is received.
+		// is slow to update, we add the namespace into a force live lookup list to ensure
+		// we are not looking at stale state.
 		if a.GetOperation() == admission.Delete {
-			l.store.Delete(&api.Namespace{
-				ObjectMeta: api.ObjectMeta{
-					Name: a.GetName(),
-				},
-			})
+			namespace := a.GetName()
+			lruItemObj, ok := l.forceLiveLookupCache.Get(namespace)
+			if !ok || lruItemObj.(forceLiveLookupEntry).expiry.Before(time.Now()) {
```

Comment:
You just received a fresh delete.  Seems like this should be added unconditionally.